src/main/java/seedu/address/model/student/NameContainsKeywordsPredicate.java

```diff
@@ -18,8 +17,9 @@ public NameContainsKeywordsPredicate(List<String> keywords) {
 
     @Override
     public boolean test(Student student) {
+        String lowercaseName = student.getName().fullName.toLowerCase();
         return keywords.stream()
-                .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(student.getName().fullName, keyword));
+                .anyMatch(keyword -> lowercaseName.contains(keyword.toLowerCase()));
```

Comment:
Great that you've created "lowercaseName" as a variable. This helps to improve the readability of the code.